Mage.Sets/src/mage/cards/n/NanogeneConversion.java

```diff
@@ -0,0 +1,76 @@
+package mage.cards.n;
+
+import mage.MageObject;
+import mage.abilities.Ability;
+import mage.abilities.effects.OneShotEffect;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.Duration;
+import mage.constants.Outcome;
+import mage.constants.SuperType;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.target.common.TargetControlledCreaturePermanent;
+import mage.util.functions.CopyApplier;
+
+import java.util.UUID;
+
+/**
+ *
+ * @author notgreat
+ */
+public final class NanogeneConversion extends CardImpl {
+
+    public NanogeneConversion(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.SORCERY}, "{3}{U}");
+
+        // Choose target creature you control. Each other creature becomes a copy of that creature until end of turn, except it isn't legendary.
+        this.getSpellAbility().addEffect(new NanogeneConversionEffect());
+        this.getSpellAbility().addTarget(new TargetControlledCreaturePermanent());
+    }
+
+    private NanogeneConversion(final NanogeneConversion card) {
+        super(card);
+    }
+
+    @Override
+    public NanogeneConversion copy() {
+        return new NanogeneConversion(this);
+    }
+}
+//Exact copy of Augmenter Pugilist's EchoingEquationEffect
```

Comment:
it's not the exact same effect - the other one is limited to creatures you control, but this isn't

I'd prefer a common class `EachOtherBecomesCopyOfTargetEffect(FilterPermanent filter, Duration duration, CopyApplier applier)`

Reply:
That'd probably be best, but for now I'll just use `getActivePermanents` instead of `getAllActivePermanents`. That whole area should probably be refactored at some point, it's weird how the former's player ID is for range of influence whereas the latter is a controller filter.